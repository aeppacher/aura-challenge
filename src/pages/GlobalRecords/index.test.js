import { configure, shallow, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { useSelector, useStore } from "react-redux";
import toJson from "enzyme-to-json";
import { GlobalRecords } from ".";
import * as React from "react";
import { mockFetch } from "../../util/mockFetch";

configure({ adapter: new Adapter() });

jest.mock("../../util/mockFetch", () => ({
  mockFetch: jest.fn(),
}));

jest.mock("react-redux", () => ({
  useStore: jest.fn(),
  useSelector: jest.fn(() => [
    {
      id: 10,
      artist: {
        name: "Jeff Parker",
        id: 100,
      },
      album: {
        title: "Suite for Max Brown",
        imageSrc: "/assets/jeff_parker-suite_for_max_brown.jpg",
      },
    },
  ]),
}));

describe("GlobalRecords component tests", () => {
  beforeEach(() => jest.clearAllMocks());

  it("should match snapshot", () => {
    const Wrapper = shallow(<GlobalRecords />);
    expect(toJson(Wrapper)).toMatchSnapshot();
  });

  it("should render and invoke useEffect once", () => {
    const useEffectMock = jest.fn();

    jest.spyOn(React, "useEffect").mockImplementation(useEffectMock);

    const Wrapper = shallow(<GlobalRecords />);
    expect(toJson(Wrapper)).toMatchSnapshot();
    expect(useEffectMock).toHaveBeenCalledTimes(1);
  });
});
