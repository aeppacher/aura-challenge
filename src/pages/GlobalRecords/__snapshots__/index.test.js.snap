// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalRecords component tests should match snapshot 1`] = `
<div
  className="aura-page aura-global_records"
>
  <h1>
    Top Records of 2020
  </h1>
  <div
    className="aura-page-content"
  />
</div>
`;

exports[`GlobalRecords component tests should render and invoke useEffect once 1`] = `
<div
  className="aura-page aura-global_records"
>
  <h1>
    Top Records of 2020
  </h1>
  <div
    className="aura-page-content"
  />
</div>
`;
